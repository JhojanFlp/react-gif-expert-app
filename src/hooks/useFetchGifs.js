import { useEffect, useState } from "react"
import { getGif } from "../helpers/gifService"

export const useFetchGifs = (category = '') => {
  const [state, setState] = useState({
    data: [],
    loading: true
  })

  // useEffect is not async - must return a cleanup function or nothing
  useEffect(() => {
    // setState({...state, loading: true})
    getGif(category)
      .then(gifs => setState({
        data: gifs,
        loading: false
      }))
  }, [category])

  return state;
}