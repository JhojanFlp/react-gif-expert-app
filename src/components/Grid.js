import React from 'react'
import PropTypes from 'prop-types'
import Gif from './Gif';

import { useFetchGifs } from '../hooks/useFetchGifs';


const Grid = ({ category }) => {

  const { data: gifs, loading } = useFetchGifs(category);
  
  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading && <p className="animate__animated animate__flash">Loading...</p>}
      <div className="grid">
        {
          gifs.map(gif => (
            <Gif 
              key={gif.id}
              {...gif}
            />
          ))
        }
      </div>
    </>
  )
}

Grid.propTypes = {
  category: PropTypes.string.isRequired
}

export default Grid
