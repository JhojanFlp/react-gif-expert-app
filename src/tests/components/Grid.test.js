import React from 'react';
import { shallow } from "enzyme";
import '@testing-library/jest-dom';

import Grid from '../../components/Grid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Test in Grid', () => {
  const category = 'Kimetsu';
  // const wrapper = shallow(<Grid category={category} />);
  test('should render component', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<Grid category={category} />);
    expect(wrapper).toMatchSnapshot();
  })
  test('should render component with gifs', () => {
    const gifs = [{ id: "1", url: "https://example.com/gif.gif" , title: "Título"}];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<Grid category={category} />);
    expect(wrapper).toMatchSnapshot();
  })
  test('should render component with information', () => {
    const gifs = [{ id: "1", url: "https://example.com/gif.gif" , title: "Título"}];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<Grid category={category} />);
    // No tag p with loading...
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('Gif').length).toBe(gifs.length);
  })
})
