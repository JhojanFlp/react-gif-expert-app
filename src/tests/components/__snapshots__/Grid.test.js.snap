// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Grid should render component 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Kimetsu
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="grid"
  />
</Fragment>
`;

exports[`Test in Grid should render component with gifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Kimetsu
  </h3>
  <div
    className="grid"
  >
    <Gif
      id="1"
      key="1"
      title="TÃ­tulo"
      url="https://example.com/gif.gif"
    />
  </div>
</Fragment>
`;
